tests/http/federation/test_matrix_federation_agent.py

```diff
@@ -719,10 +721,12 @@ def test_get_well_known_unsigned_cert(self):
         agent = MatrixFederationAgent(
             reactor=self.reactor,
             tls_client_options_factory=tls_factory,
+            user_agent=b"test-agent",  # This is unused since _well_known_resolver is passed below.
             _srv_resolver=self.mock_resolver,
             _well_known_resolver=WellKnownResolver(
                 self.reactor,
                 Agent(self.reactor, contextFactory=tls_factory),
+                b"test-agent",
```

Comment:
It doesn't look like we're asserting that the user agent appears on the other side anywhere?

Reply:
I added a couple of assertions, hopefully that fits what you were looking for! (I didn't add it specifically for this test, but some other ones.)

Reply:
Looks good, thanks!